### PR TITLE
Fixes #13634 - Adding Katello cert to ca-trust

### PR DESCRIPTION
**Pull request by parthaa**
Adding Katello Server cert to the machines local ca-trust

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -129,6 +129,7 @@
 
   $katello_server_ca_cert = "${certs::pki_dir}/certs/${server_ca_name}.crt"
 
+  class { 'trusted_ca': } ->
```
Comment (ekohl) [inline manifests/init.pp]: This should be ```::trusted_ca```

**Comment by ekohl** [comment]
Dependencies should also be in .fixtures.yml, preferably their git versions so we can spot failures sooner.

File: metadata.json [hunk 2]
```diff
@@ -9,6 +9,10 @@
   "issues_url": "http://projects.theforeman.org/projects/katello/issues",
   "dependencies": [
     {
+      "name": "evenup-trusted_ca",
+      "version_requirement": ">= 1.0.1"
```
Comment (ehelms) [inline metadata.json]: Please add `< 2.0.0` to the end to help us avoid major version breakages in the future.

File: manifests/katello.pp [hunk 3]
```diff
@@ -18,6 +18,12 @@
   $candlepin_consumer_summary     = "Subscription-manager consumer certificate for Katello instance ${::fqdn}"
   $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
 
+  class { 'trusted_ca': }
```
Comment (ehelms) [inline manifests/katello.pp]: Is this needed if you are using the defined type? If it is needed, can you use include instead?

File: manifests/katello.pp [hunk 4]
```diff
@@ -18,6 +18,12 @@
   $candlepin_consumer_summary     = "Subscription-manager consumer certificate for Katello instance ${::fqdn}"
   $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
 
+  class { ::trusted_ca }
+  trusted_ca::ca { 'katello_server-host-cert':
+    source => "${certs::pki_dir}/certs/${certs::server_ca_name}.crt",
+    require => File["${certs::pki_dir}/certs/${certs::server_ca_name}.crt"],
```
Comment (ehelms) [inline manifests/katello.pp]: Is it able to resolve `$::certs::katello_server_ca_cert` here?

**Comment by parthaa** [comment]
> Dependencies should also be in .fixtures.yml, preferably their git versions so we can spot failures sooner.

thanks will look into that

File: manifests/katello.pp (hunk 4 above)
Comment (parthaa) [inline manifests/katello.pp]: Will test your suggestion. But I got the variables from 2 lines below this.

File: .fixtures.yml [hunk 5]
```diff
@@ -4,5 +4,6 @@ fixtures:
     extlib:        "git://github.com/puppet-community/puppet-extlib.git"
     foreman:       "git://github.com/theforeman/puppet-foreman.git"
     common:        "git://github.com/katello/puppet-common.git"
+    evenup:        "git@github.com:evenup/evenup-trusted_ca.git"
```
Comment (ekohl) [inline .fixtures.yml]: Please use ```git://``` because it's anonymous.

File: manifests/katello.pp [hunk 6]
```diff
@@ -18,6 +18,12 @@
   $candlepin_consumer_summary     = "Subscription-manager consumer certificate for Katello instance ${::fqdn}"
   $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
 
+  class { '::trusted_ca': } ->
+  trusted_ca::ca { 'katello_server-host-cert':
+    source => $certs::katello_server_ca_cert,
```
Comment (ekohl) [inline manifests/katello.pp]: Please align the parameters.

**Comment by parthaa** [comment]
> thanks will look into that

@ekohl  added

File: .fixtures.yml [hunk 7]
```diff
@@ -4,5 +4,6 @@ fixtures:
     extlib:        "git://github.com/puppet-community/puppet-extlib.git"
     foreman:       "git://github.com/theforeman/puppet-foreman.git"
     common:        "git://github.com/katello/puppet-common.git"
+    evenup:        "git://github.com/evenup/evenup-trusted_ca.git"
```
Comment (ehelms) [inline .fixtures.yml]: Ideally the name here would be the name of the puppet module and not the provider of the module, i.e. trusted_ca and not evenup

File: manifests/katello.pp [hunk 8]
```diff
@@ -18,6 +18,12 @@
   $candlepin_consumer_summary     = "Subscription-manager consumer certificate for Katello instance ${::fqdn}"
   $candlepin_consumer_description = 'Consumer certificate and post installation script that configures rhsm.'
 
+  class { '::trusted_ca': }
```
Comment (ehelms) [inline manifests/katello.pp]: Can this be done via `include ::trustedca` since we aren't setting any parameters?

**Comment by parthaa** [comment]
Also tested this the following way

- Installed katello-devel
- saw this file ```/etc/pki/ca-trust/source/anchors/katello_server-host-cert.crt```
- tried ```curl https://<fqdn>``` . It responded with a 503 but not a 401, insecure.
- I then did
```$ rm -f /etc/pki/ca-trust/source/anchors/katello_server-host-cert.crt
   $ update-ca-trust
```
- Again tried ```curl https://<fqdn>``` 
- It responded with a 401, insecure error.

File: manifests/katello.pp (hunk 8 above)
Comment (parthaa) [inline manifests/katello.pp]: Is there a precedent for this ?

Comment (parthaa) [inline manifests/katello.pp]: will try

File: spec/classes/katello_spec.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+require 'spec_helper'
```
Comment (ehelms) [inline spec/classes/katello_spec.rb]: Nitpick: typically the test name follows the full manifest name as a convention, i.e. `certs_katello_spec.rb`

**Comment by ehelms** [comment]
Just to double check, we want this to be deployed on both the main server and all Capsules is that correct? Currently, the capsule uses the `katello.pp` manifest -- https://github.com/Katello/puppet-capsule/blob/f05a17386fc7575d7944ea752853696af3ebf6d3/manifests/init.pp#L104

**Comment by parthaa** [comment]
> Just to double check, we want this to be deployed on both the main server and all Capsules is that correct? Currently, the capsule uses the katello.pp manifest -- https://github.com/Katello/puppet-capsule/blob/f05a17386fc7575d7944ea752853696af3ebf6d3/manifests/init.pp#L104

Cert should probably be trusted only the main server for post sync to work afaik.  @jlsherrill with the nodes work do think this will be useful in a capsule?  
One thing running this in a capsule would be if customer wants to turn that capsule into a Docker node.

**Comment by ehelms** [comment]
ACK - thanks @parthaa !

**Pull request merged**